crates/mithril-oxide-sys/cpp/IR/BuiltinTypes.cpp

```diff
@@ -0,0 +1,24 @@
+#include "BuiltinTypes.hpp"
+
+#include <memory>
+#include <mlir/IR/BuiltinAttributes.h>
+
+namespace mithril_oxide_sys
+{
+
+std::unique_ptr<IntegerType> IntegerType_get(MLIRContext &context, unsigned int width, bool has_sign, bool is_signed)
+{
+    IntegerType::SignednessSemantics semantics;
+    if(has_sign)
+    {
+        semantics = is_signed ? IntegerType::SignednessSemantics::Signed : IntegerType::SignednessSemantics::Unsigned;
+    }
+    else
+    {
+        semantics = IntegerType::SignednessSemantics::Signless;
+    }
```

Comment:
Usually, if I have if-else statements with a single simple statement I omit the braces, like this:
```suggestion
    if(has_sign)
        semantics = is_signed
            ? IntegerType::SignednessSemantics::Signed
            : IntegerType::SignednessSemantics::Unsigned;
    else
        semantics = IntegerType::SignednessSemantics::Signless;
```